dask/array/percentile.py

```diff
@@ -12,24 +12,24 @@
 
 @wraps(np.percentile)
 def _percentile(a, q, interpolation="linear"):
-    import pandas as pd
-
     n = len(a)
     if not len(a):
         return None, n
     if isinstance(q, Iterator):
         q = list(q)
     if a.dtype.name == "category":
         result = np.percentile(a.cat.codes, q, interpolation=interpolation)
+        import pandas as pd
 
         return pd.Categorical.from_codes(result, a.dtype.categories, a.dtype.ordered), n
-    if isinstance(a.dtype, pd.core.dtypes.dtypes.DatetimeTZDtype) or np.issubdtype(
-        a.dtype, np.datetime64
-    ):
+    if type(a.dtype).__name__ == "DatetimeTZDtype":
+        import pandas as pd
+
         if isinstance(a, (pd.Series, pd.Index)):
-            values = a.values
-        else:
-            values = a
+            a = a.values
```

Comment:
Would it make sense to throw these into an `else` where we do the `import` and then handle different cases?

Reply:
I think it would result in duplication of `datetime` related handling logic as that is needed both for pandas and numpy case.

Reply:
Yeah just trying to see if there is a good way to avoid the `__name__` check and keep the `isinstance` check

Reply:
Any other thoughts on this Prem?

Reply:
Can't think of approaches other than the one in PR and try/excpet here.